backend/src/main/java/atlantbh/restaurants/services/LocationService.java

```diff
@@ -0,0 +1,45 @@
+package atlantbh.restaurants.services;
+
+import atlantbh.restaurants.models.Location;
+import atlantbh.restaurants.models.filters.LocationFilterBuilder;
+import atlantbh.restaurants.models.sortkeys.LocationSortKeys;
+import atlantbh.restaurants.repositories.LocationRepository;
+import org.hibernate.Criteria;
+import org.hibernate.criterion.Restrictions;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+public class LocationService extends BaseService<Location, LocationSortKeys, LocationFilterBuilder, LocationRepository> {
+
+    // TODO Change method to use Criteria instead of native query
+    public List<Location> findAllCitiesForCountry(String country) {
+        Query query = repository.getSession().createNativeQuery("SELECT *  FROM Location l WHERE l.id IN " +
```

Comment:
service should not do this.. pass to repository

Reply:
Fixed 8c841a20e489a31c66d177c8682339f9d4f19de0